samples/snippets/update_with_dml.py

```diff
@@ -60,10 +62,13 @@ def update_with_dml(
     query_job.result()
 
     print(f"DML query modified {query_job.num_dml_affected_rows} rows.")
-    return query_job.num_dml_affected_rows
+    return typing.cast(int, query_job.num_dml_affected_rows)
```

Comment:
I assume this is because `num_dml_affected_rows` is `Optional`? Could we change the function return type instead and avoid the cast?

Reply:
Yes, that's the reason that `mypy` (rightfully) complains.

Since `num_dml_affected_rows` can actually be `None`,  I don't think we can just change the return type. But there is another way, possibly better - if we explicitly assert that the target value is not `None`, mypy will understand it, and the cast will not be needed anymore.

```py
assert query_job.num_dml_affected_rows is not None
return query_job.num_dml_affected_rows  # passes the type check
```